test/torchaudio_unittest/prototype/functional/functional_test_impl.py

```diff
@@ -112,6 +112,263 @@ def test_add_noise_length_check(self):
+        """Check that when the value of absorption and scattering are the same
+        across walls and frequency bands, the output histograms are:
+        - all equal across frequency bands
+        - equal to simply passing a float value instead of a (num_bands, D) or
+        (D,) tensor.
+        """
+
+        room_dim = torch.tensor([20, 25], dtype=self.dtype)
+        mic_array = torch.tensor([[2, 2], [8, 8]], dtype=self.dtype)
+        source = torch.tensor([7, 6], dtype=self.dtype)
+        num_rays = 1_000
+        ABS, SCAT = 0.1, 0.2
+
+        e_absorption = torch.full(fill_value=ABS, size=(6, 4), dtype=self.dtype)
+        scattering = torch.full(fill_value=SCAT, size=(6, 4), dtype=self.dtype)
+        hist_per_band_per_wall = F.ray_tracing(
+            room=room_dim,
+            source=source,
+            mic_array=mic_array,
+            num_rays=num_rays,
+            e_absorption=e_absorption,
+            scattering=scattering,
+        )
+        e_absorption = torch.full(fill_value=ABS, size=(4,), dtype=self.dtype)
+        scattering = torch.full(fill_value=SCAT, size=(4,), dtype=self.dtype)
+        hist_per_wall = F.ray_tracing(
+            room=room_dim,
+            source=source,
+            mic_array=mic_array,
+            num_rays=num_rays,
+            e_absorption=e_absorption,
+            scattering=scattering,
+        )
+
+        e_absorption = ABS
+        scattering = SCAT
+        hist_single = F.ray_tracing(
+            room=room_dim,
+            source=source,
+            mic_array=mic_array,
+            num_rays=num_rays,
+            e_absorption=e_absorption,
+            scattering=scattering,
+        )
+        assert hist_per_band_per_wall.shape == (2, 6, 2500)
+        assert hist_per_wall.shape == (2, 1, 2500)
+        assert hist_single.shape == (2, 1, 2500)
+        torch.testing.assert_close(hist_single, hist_per_wall)
+
+        hist_single = hist_single.expand(2, 6, 2500)
+        torch.testing.assert_close(hist_single, hist_per_band_per_wall)
+
+    @parameterized.expand(
+        [
+            ([20, 25], [2, 2], [[8, 8], [7, 6]], 10_000, 5.563379765),  # 2D with 2 mics
+            ([20, 25, 30], [1, 10, 5], [[8, 8, 22]], 1_000, 0.038723841),  # 3D with 1 mic
+        ]
+    )
+    def test_ray_tracing_same_results_as_pyroomacoustics(self, room_dim, source, mic_array, num_rays, expected_sum):
+        """Ideally we would be checking the histogram directly against
+        pyroomacoustics, but unfortunately PRA may randomly segfault during the
+        ray-tracing. So instead we check the overlall sum of the histrogram
+        values.
+
+        The sums have been validated *on each bin* against PRA with high
+        precision, i.e. this was passing:
+
+        self.assertEqual(hist, hist_pra)  # check all bin values with default atol and rtol
+        Note: need to set max_order=0 and air_absorption=False in PRA for consistent checks.
```

Comment:
`max_order=0` means without image source method in pyroomacoustics
could you add explanation in the comment so that maintainers know what's the meaning of the settings?